File: primes.jl
function is_prime(n::Int64)
    if n < 4
        return true
    end
    if n % 2 == 0
        return false
    end
    for a = 3:round(sqrt(n))
        if n % a == 0
            return false
        end
    end
    return true
end

function calculate_primes(n::Int64)
    primes = Int64[]
    p = 1
    for i = 1:n
        while ! is_prime(p)
            p += 1
        end
        push!(primes, p)
        p += 1
    end
    return primes
end

calculate_primes(100000)

File: primes.cpp
#include <cmath>
#include <vector>
#include <iostream>

int is_prime(int n)
{
  if (n < 4)
    return 1;
  if (!(n % 2))
    return 0;
  for (int a = 3; a < (int)(sqrt(n)) + 1; a++)
    if (!(n % a))
      return 0;
  return 1;
}

std::vector<int> calculate_primes(int n)
{
  std::vector<int> primes;
  int p = 1;
  for (int i = 0; i < n; i++) {
    while (!is_prime(p))
      p++;
    primes.push_back(p);
    p++;
  }

  return primes;
}

int main()
{
  calculate_primes(1000000);

  return 0;
}

